homeassistant/components/media_player/dlna_dmr.py

```diff
@@ -0,0 +1,542 @@
+    from async_upnp_client import UpnpRequester
+
+    class HassUpnpRequester(UpnpRequester):
+        """async_upnp_client.UpnpRequester for home-assistant."""
+
+        def __init__(self, hass_):
+            """Initializer."""
+            self.hass = hass_
+
+        async def async_do_http_request(self,
+                                        method,
+                                        url,
+                                        headers=None,
+                                        body=None,
+                                        body_type='text'):
+            """Do a HTTP request."""
+            # work around an unknown hass or aiohttp error
+            await asyncio.sleep(0.01)
+
+            session = async_get_clientsession(self.hass)
+            with async_timeout.timeout(5, loop=self.hass.loop):
+                response = await session.request(method,
+                                                 url,
+                                                 headers=headers,
+                                                 data=body)
+                if body_type == 'text':
+                    response_body = await response.text()
+                elif body_type == 'raw':
+                    response_body = await response.read()
+                elif body_type == 'ignore':
+                    response_body = None
+
+            return response.status, response.headers, response_body
+
+    return HassUpnpRequester(hass)
+
+
+class DlnaDmrDevice(MediaPlayerDevice):
+    """Representation of a DLNA DMR device."""
+
+    def __init__(self, hass: HomeAssistant, dmr_device, name=None):
+        """Initializer."""
+        self.hass = hass
+        self._name = name
+
+        self._device = dmr_device
+        self._device.on_event = self._on_event
+
+        self._available = False
+        self._on_stop_unsubscriber = None
+        self._subscription_renew_time = None
+
+    async def async_added_to_hass(self):
+        """Callback when added."""
+        bus = self.hass.bus
+
+        # register unsubscribe on stop
+        bus.async_listen_once(EVENT_HOMEASSISTANT_STOP,
+                              self._async_on_hass_stop)
+
+    @property
+    def udn(self):
+        """Get UDN of DLNA DMR device."""
+        return self._device.udn
+
+    @property
+    def available(self):
+        """Device is available."""
+        return self._available
+
+    async def _async_on_hass_stop(self, event):
+        """Event handler on HASS stop."""
+        await self._device.async_unsubscribe_services()
+
+    async def async_update(self):
+        """Retrieve the latest data."""
+        was_available = self._available
+
+        try:
+            await self._device.async_update()
+            self._available = True
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            self._available = False
+            _LOGGER.debug("Device unavailable")
+            raise
```

Comment:
Don't raise. Just log an error and return. Replace the debug log with the error log.